src/test/java/seedu/address/logic/parser/FindTransactionCommandParserTest.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.logic.parser.FindTransactionCommandParser.TransactionType.BUY_TRANSACTION;
+import static seedu.address.logic.parser.FindTransactionCommandParser.TransactionType.EMPTY;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.FindTransactionCommand;
+import seedu.address.model.transaction.TransactionContainKeywordsPredicate;
+
+public class FindTransactionCommandParserTest {
+
+    private FindTransactionCommandParser parser = new FindTransactionCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "      ", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                FindTransactionCommand.MESSAGE_NO_FIELD_PROVIDED));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFindCommand() {
+        // no leading and trailing whitespaces
+        FindTransactionCommand expectedFindTransactionCommand =
+                new FindTransactionCommand(new TransactionContainKeywordsPredicate(BUY_TRANSACTION,
+                        Arrays.asList("Alice", "Bob"), Arrays.asList("Apple", "Banana")));
+        assertParseSuccess(parser, "buy n/Alice Bob g/Apple Banana", expectedFindTransactionCommand);
+
+        // multiple whitespaces between keywords
+        assertParseSuccess(parser, "buy n/\n Alice \n \t Bob  \t g/Apple Banana",
+                expectedFindTransactionCommand);
```

Comment:
There is a CommandTestUtil.PREAMBLE_WHITESPACE, not sure if you would find that helpful.

Reply:
dint know that got such variable present, i will use it